File: kmp/composeApp/src/commonMain/kotlin/com/bloom/core/data/GrowthData.kt
package com.bloom.core.data

data class GrowthPoint(val month: Int, val p3: Double, val p50: Double, val p97: Double)

val kWeightBoys = listOf(
    GrowthPoint(0,  2.5,  3.3,  4.4),
    GrowthPoint(1,  3.4,  4.5,  5.8),
    GrowthPoint(2,  4.4,  5.6,  7.1),
    GrowthPoint(3,  5.1,  6.4,  8.0),
    GrowthPoint(4,  5.7,  7.0,  8.7),
    GrowthPoint(5,  6.1,  7.5,  9.3),
    GrowthPoint(6,  6.4,  7.9,  9.7),
    GrowthPoint(7,  6.7,  8.3,  10.2),
    GrowthPoint(8,  7.0,  8.6,  10.5),
    GrowthPoint(9,  7.2,  8.9,  10.9),
    GrowthPoint(10, 7.5,  9.2,  11.2),
    GrowthPoint(11, 7.7,  9.4,  11.5),
    GrowthPoint(12, 7.8,  9.6,  11.8),
    GrowthPoint(15, 8.3,  10.3, 12.6),
    GrowthPoint(18, 8.8,  10.9, 13.4),
    GrowthPoint(21, 9.2,  11.5, 14.1),
    GrowthPoint(24, 9.7,  12.2, 15.0),
)

val kWeightGirls = listOf(
    GrowthPoint(0,  2.4,  3.2,  4.2),
    GrowthPoint(1,  3.2,  4.2,  5.5),
    GrowthPoint(2,  3.9,  5.1,  6.6),
    GrowthPoint(3,  4.5,  5.8,  7.5),
    GrowthPoint(4,  5.0,  6.4,  8.2),
    GrowthPoint(5,  5.4,  6.9,  8.8),
    GrowthPoint(6,  5.7,  7.3,  9.3),
    GrowthPoint(7,  6.0,  7.6,  9.8),
    GrowthPoint(8,  6.3,  7.9,  10.2),
    GrowthPoint(9,  6.5,  8.2,  10.5),
    GrowthPoint(10, 6.7,  8.5,  10.9),
    GrowthPoint(11, 6.9,  8.7,  11.2),
    GrowthPoint(12, 7.1,  8.9,  11.5),
    GrowthPoint(15, 7.6,  9.6,  12.4),
    GrowthPoint(18, 8.1,  10.2, 13.2),
    GrowthPoint(21, 8.6,  10.9, 14.0),
    GrowthPoint(24, 9.0,  11.5, 14.8),
)

val kHeightBoys = listOf(
    GrowthPoint(0,  46.3, 49.9, 53.4),
    GrowthPoint(1,  50.8, 54.7, 58.6),
    GrowthPoint(2,  54.4, 58.4, 62.4),
    GrowthPoint(3,  57.3, 61.4, 65.5),
    GrowthPoint(4,  59.7, 63.9, 68.0),
    GrowthPoint(5,  61.7, 65.9, 70.1),
    GrowthPoint(6,  63.3, 67.6, 71.9),
    GrowthPoint(7,  64.8, 69.2, 73.5),
    GrowthPoint(8,  66.2, 70.6, 75.0),
    GrowthPoint(9,  67.5, 72.0, 76.5),
    GrowthPoint(10, 68.7, 73.3, 77.9),
    GrowthPoint(11, 69.9, 74.5, 79.2),
    GrowthPoint(12, 71.0, 75.7, 80.5),
    GrowthPoint(15, 73.9, 79.1, 84.2),
    GrowthPoint(18, 76.9, 82.3, 87.7),
    GrowthPoint(21, 79.4, 85.1, 90.7),
    GrowthPoint(24, 81.7, 87.8, 93.9),
)

val kHeightGirls = listOf(
    GrowthPoint(0,  45.6, 49.1, 52.7),
    GrowthPoint(1,  49.8, 53.7, 57.6),
    GrowthPoint(2,  53.0, 57.1, 61.1),
    GrowthPoint(3,  55.6, 59.8, 64.0),
    GrowthPoint(4,  57.8, 62.1, 66.4),
    GrowthPoint(5,  59.6, 64.0, 68.5),
    GrowthPoint(6,  61.2, 65.7, 70.3),
    GrowthPoint(7,  62.7, 67.3, 71.9),
    GrowthPoint(8,  64.0, 68.7, 73.5),
    GrowthPoint(9,  65.3, 70.1, 75.0),
    GrowthPoint(10, 66.5, 71.5, 76.4),
    GrowthPoint(11, 67.7, 72.8, 77.8),
    GrowthPoint(12, 68.9, 74.0, 79.2),
    GrowthPoint(15, 72.0, 77.5, 83.1),
    GrowthPoint(18, 74.9, 80.7, 86.5),
    GrowthPoint(21, 77.5, 83.7, 89.9),
    GrowthPoint(24, 80.0, 86.4, 92.9),
)

fun getGrowthData(gender: String, isWeight: Boolean): List<GrowthPoint> =
    if (isWeight) if (gender == "boy") kWeightBoys else kWeightGirls
    else if (gender == "boy") kHeightBoys else kHeightGirls

File: lib/core/data/growth_data.dart
class GrowthPoint {
  final int month;
  final double p3;
  final double p50;
  final double p97;

  const GrowthPoint({
    required this.month,
    required this.p3,
    required this.p50,
    required this.p97,
  });
}

// WHO 2006 — Poids pour l'âge, garçons (kg)
const List<GrowthPoint> kWeightBoys = [
  GrowthPoint(month: 0,  p3: 2.5, p50: 3.3,  p97: 4.4),
  GrowthPoint(month: 1,  p3: 3.4, p50: 4.5,  p97: 5.8),
  GrowthPoint(month: 2,  p3: 4.4, p50: 5.6,  p97: 7.1),
  GrowthPoint(month: 3,  p3: 5.1, p50: 6.4,  p97: 8.0),
  GrowthPoint(month: 4,  p3: 5.7, p50: 7.0,  p97: 8.7),
  GrowthPoint(month: 5,  p3: 6.1, p50: 7.5,  p97: 9.3),
  GrowthPoint(month: 6,  p3: 6.4, p50: 7.9,  p97: 9.7),
  GrowthPoint(month: 7,  p3: 6.7, p50: 8.3,  p97: 10.2),
  GrowthPoint(month: 8,  p3: 7.0, p50: 8.6,  p97: 10.5),
  GrowthPoint(month: 9,  p3: 7.2, p50: 8.9,  p97: 10.9),
  GrowthPoint(month: 10, p3: 7.5, p50: 9.2,  p97: 11.2),
  GrowthPoint(month: 11, p3: 7.7, p50: 9.4,  p97: 11.5),
  GrowthPoint(month: 12, p3: 7.8, p50: 9.6,  p97: 11.8),
  GrowthPoint(month: 15, p3: 8.3, p50: 10.3, p97: 12.6),
  GrowthPoint(month: 18, p3: 8.8, p50: 10.9, p97: 13.4),
  GrowthPoint(month: 21, p3: 9.2, p50: 11.5, p97: 14.1),
  GrowthPoint(month: 24, p3: 9.7, p50: 12.2, p97: 15.0),
];

// WHO 2006 — Poids pour l'âge, filles (kg)
const List<GrowthPoint> kWeightGirls = [
  GrowthPoint(month: 0,  p3: 2.4, p50: 3.2,  p97: 4.2),
  GrowthPoint(month: 1,  p3: 3.2, p50: 4.2,  p97: 5.5),
  GrowthPoint(month: 2,  p3: 3.9, p50: 5.1,  p97: 6.6),
  GrowthPoint(month: 3,  p3: 4.5, p50: 5.8,  p97: 7.5),
  GrowthPoint(month: 4,  p3: 5.0, p50: 6.4,  p97: 8.2),
  GrowthPoint(month: 5,  p3: 5.4, p50: 6.9,  p97: 8.8),
  GrowthPoint(month: 6,  p3: 5.7, p50: 7.3,  p97: 9.3),
  GrowthPoint(month: 7,  p3: 6.0, p50: 7.6,  p97: 9.8),
  GrowthPoint(month: 8,  p3: 6.3, p50: 7.9,  p97: 10.2),
  GrowthPoint(month: 9,  p3: 6.5, p50: 8.2,  p97: 10.5),
  GrowthPoint(month: 10, p3: 6.7, p50: 8.5,  p97: 10.9),
  GrowthPoint(month: 11, p3: 6.9, p50: 8.7,  p97: 11.2),
  GrowthPoint(month: 12, p3: 7.1, p50: 8.9,  p97: 11.5),
  GrowthPoint(month: 15, p3: 7.6, p50: 9.6,  p97: 12.4),
  GrowthPoint(month: 18, p3: 8.1, p50: 10.2, p97: 13.2),
  GrowthPoint(month: 21, p3: 8.6, p50: 10.9, p97: 14.0),
  GrowthPoint(month: 24, p3: 9.0, p50: 11.5, p97: 14.8),
];

// WHO 2006 — Taille pour l'âge, garçons (cm)
const List<GrowthPoint> kHeightBoys = [
  GrowthPoint(month: 0,  p3: 46.3, p50: 49.9, p97: 53.4),
  GrowthPoint(month: 1,  p3: 50.8, p50: 54.7, p97: 58.6),
  GrowthPoint(month: 2,  p3: 54.4, p50: 58.4, p97: 62.4),
  GrowthPoint(month: 3,  p3: 57.3, p50: 61.4, p97: 65.5),
  GrowthPoint(month: 4,  p3: 59.7, p50: 63.9, p97: 68.0),
  GrowthPoint(month: 5,  p3: 61.7, p50: 65.9, p97: 70.1),
  GrowthPoint(month: 6,  p3: 63.3, p50: 67.6, p97: 71.9),
  GrowthPoint(month: 7,  p3: 64.8, p50: 69.2, p97: 73.5),
  GrowthPoint(month: 8,  p3: 66.2, p50: 70.6, p97: 75.0),
  GrowthPoint(month: 9,  p3: 67.5, p50: 72.0, p97: 76.5),
  GrowthPoint(month: 10, p3: 68.7, p50: 73.3, p97: 77.9),
  GrowthPoint(month: 11, p3: 69.9, p50: 74.5, p97: 79.2),
  GrowthPoint(month: 12, p3: 71.0, p50: 75.7, p97: 80.5),
  GrowthPoint(month: 15, p3: 73.9, p50: 79.1, p97: 84.2),
  GrowthPoint(month: 18, p3: 76.9, p50: 82.3, p97: 87.7),
  GrowthPoint(month: 21, p3: 79.4, p50: 85.1, p97: 90.7),
  GrowthPoint(month: 24, p3: 81.7, p50: 87.8, p97: 93.9),
];

// WHO 2006 — Taille pour l'âge, filles (cm)
const List<GrowthPoint> kHeightGirls = [
  GrowthPoint(month: 0,  p3: 45.6, p50: 49.1, p97: 52.7),
  GrowthPoint(month: 1,  p3: 49.8, p50: 53.7, p97: 57.6),
  GrowthPoint(month: 2,  p3: 53.0, p50: 57.1, p97: 61.1),
  GrowthPoint(month: 3,  p3: 55.6, p50: 59.8, p97: 64.0),
  GrowthPoint(month: 4,  p3: 57.8, p50: 62.1, p97: 66.4),
  GrowthPoint(month: 5,  p3: 59.6, p50: 64.0, p97: 68.5),
  GrowthPoint(month: 6,  p3: 61.2, p50: 65.7, p97: 70.3),
  GrowthPoint(month: 7,  p3: 62.7, p50: 67.3, p97: 71.9),
  GrowthPoint(month: 8,  p3: 64.0, p50: 68.7, p97: 73.5),
  GrowthPoint(month: 9,  p3: 65.3, p50: 70.1, p97: 75.0),
  GrowthPoint(month: 10, p3: 66.5, p50: 71.5, p97: 76.4),
  GrowthPoint(month: 11, p3: 67.7, p50: 72.8, p97: 77.8),
  GrowthPoint(month: 12, p3: 68.9, p50: 74.0, p97: 79.2),
  GrowthPoint(month: 15, p3: 72.0, p50: 77.5, p97: 83.1),
  GrowthPoint(month: 18, p3: 74.9, p50: 80.7, p97: 86.5),
  GrowthPoint(month: 21, p3: 77.5, p50: 83.7, p97: 89.9),
  GrowthPoint(month: 24, p3: 80.0, p50: 86.4, p97: 92.9),
];

List<GrowthPoint> getGrowthData({required String gender, required bool isWeight}) {
  if (isWeight) return gender == 'boy' ? kWeightBoys : kWeightGirls;
  return gender == 'boy' ? kHeightBoys : kHeightGirls;
}
